Application/Education/Certificate/Generator/Service/DataCertificate/IDataEZSH.php

```diff
@@ -0,0 +1,514 @@
+    {
+
+        $tblCertificate = $Data->createCertificate('Gymnasium Halbjahreszeugnis', 'Klasse 10',
+            'EZSH\EzshGymHjZ', $tblConsumerCertificate);
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypeGym()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeHalfYear(),
+                    $Data->getTblSchoolTypeGym());
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypeGym(), '10'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung der Einschätzung
+            $FieldName = 'Rating';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)) {
+                $Data->createCertificateField($tblCertificate, $FieldName, 2500);
+            }
+            // Begrenzung Bemerkungsfeld
+            $FieldName = 'RemarkWithoutTeam';
+            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)) {
+                $Data->createCertificateField($tblCertificate, $FieldName, 1000);
+            }
+            if (!$Data->getCertificateSubjectAll($tblCertificate)) {
+                $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+                $Data->setCertificateSubject($tblCertificate, 'EN', 1, 2);
+                $Data->setCertificateSubject($tblCertificate, 'TSCN', 1, 3, false);
+                $Data->setCertificateSubject($tblCertificate, 'LA', 1, 4, false);
+                // lücke für Fremdsprachen
+                $Data->setCertificateSubject($tblCertificate, 'KU', 1, 6);
+                $Data->setCertificateSubject($tblCertificate, 'MU', 1, 7);
+                $Data->setCertificateSubject($tblCertificate, 'GE', 1, 8);
+                $Data->setCertificateSubject($tblCertificate, 'WTH', 1, 9);
+                $Data->setCertificateSubject($tblCertificate, 'GEO', 1, 10);
+
+                $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+                $Data->setCertificateSubject($tblCertificate, 'BIO', 2, 2);
+                $Data->setCertificateSubject($tblCertificate, 'CH', 2, 3);
+                $Data->setCertificateSubject($tblCertificate, 'PH', 2, 4);
+                $Data->setCertificateSubject($tblCertificate, 'INF', 2, 5);
+                $Data->setCertificateSubject($tblCertificate, 'TUC', 2, 6);
+                $Data->setCertificateSubject($tblCertificate, 'RE/e', 2, 7);
+                $Data->setCertificateSubject($tblCertificate, 'SPO', 2, 8);
+            }
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEzshGymJ(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
Codacy found an issue: [The method setEzshGymJ() has an NPath complexity of 521. The configured NPath complexity threshold is 200.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)